### DISCUSSION TITLE
Any support for various version restrictions per branch?

### DISCUSSION THREAD

**Discussion by zeldigas** [renovatebot/renovate#8434] [Request Help] [answered]
According to docs:
> Renovate always uses the config from the repository's default branch, even if that configuration specifies multiple baseBranches. Renovate does not read/override the config from within each base branch if present.

While it works for "mainline" style of development (that is good, no doubt), looks like it's impossible to also automate dependency management for "support" branches - one that still need to get updates, but potentially with altenative restrictions on allowed versions. For example I have a java app that targets latest SpringBoot (e.g. 2.4.2) and for some compatibility reason there is still "support/prev_version" branch that should receive only fixes for prev version of framework (e.g. 2.3.x.RELEASE).

Any tricks/options to setup such configuration without moving "support" branch to separate repo?

**Comment by viceice** [comment, marked as answer]
checkout [`baseBranchList`](https://docs.renovatebot.com/configuration-options/#basebranchlist)

**Reply by rarkins** [reply]
i.e. write packageRules for any baseBranch-specific rules you need.

Renovate will read config only from the base branch until #2496 is implemented - after that you would be able to have a separate config per-base branch

**Reply by zeldigas** [reply]
Got it, thank you!
